pybamm/models/full_battery_models/base_battery_model.py

```diff
@@ -372,10 +372,35 @@ def set_voltage_variables(self):
         eta_r_av_dim = eta_r_p_av_dim - eta_r_n_av_dim
 
         # terminal voltage
-        phi_s_p = self.variables["Positive electrode potential"]
-        phi_s_p_dim = self.variables["Positive electrode potential [V]"]
-        V = pybamm.BoundaryValue(phi_s_p, "right")
-        V_dim = pybamm.BoundaryValue(phi_s_p_dim, "right")
+        if self.options["bc_options"]["dimensionality"] == 0:
+            phi_s_p = self.variables["Positive electrode potential"]
+            phi_s_p_dim = self.variables["Positive electrode potential [V]"]
+            V = pybamm.BoundaryValue(phi_s_p, "right")
+            V_dim = pybamm.BoundaryValue(phi_s_p_dim, "right")
+        elif self.options["bc_options"]["dimensionality"] == 1:
+            # TO DO: add terminal voltage in 1plus1D
+            phi_s_p = self.variables["Positive electrode potential"]
+            phi_s_p_dim = self.variables["Positive electrode potential [V]"]
+            V = pybamm.BoundaryValue(phi_s_p, "right")
+            V_dim = pybamm.BoundaryValue(phi_s_p_dim, "right")
+        elif self.options["bc_options"]["dimensionality"] == 2:
+            phi_s_cn = self.variables["Negative current collector potential"]
+            phi_s_cp = self.variables["Positive current collector potential"]
+            phi_s_cn_dim = self.variables["Negative current collector potential [V]"]
+            phi_s_cp_dim = self.variables["Positive current collector potential [V]"]
+            # In 2D left corresponds to the negative tab and right the positive tab
+            V = pybamm.BoundaryValue(phi_s_cp, "right") - pybamm.BoundaryValue(
+                phi_s_cn, "left"
+            )
+            V_dim = pybamm.BoundaryValue(phi_s_cp_dim, "right") - pybamm.BoundaryValue(
+                phi_s_cn_dim, "left"
+            )
```

Comment:
ok for now but we'll need to think of less confusing nomenclature - maybe allow "negative tab" and "positive tab" as keys in the bc dicts, but this is an issue for later

Reply:
I agree